backend/core/src/main/java/org/sonarsource/sonarlint/core/analysis/AnalysisService.java

```diff
@@ -716,6 +719,7 @@ public void didChangeAutomaticAnalysisSetting(boolean enabled) {
     if (!previouslyEnabled) {
       triggerAnalysisForOpenFiles();
     }
+    telemetryService.automaticAnalysisSettingToggled();
```

Comment:
Instead of introducing a dependency on the telemetry service, I would suggest to use the event publisher, to decouple things a bit. Telemetry is a secondary concern compared to business logic, so in most places with manage it with events, and the telemetry service reacts to those events